src/lib/test.js

```diff
@@ -74,28 +73,29 @@ const runTest = function(index) {
                       next(e)
                     })
                 },
-                //  4) test teardown (should this be in the done step to make sure it *always* runs?)
-                next => {
-                  shouldRunTeardown(this.test.repeat, index)
-                    ? runTeardown(this, this.test.teardown)
-                        .then(res => {
-                          next()
-                        })
-                        .catch(e => {
-                          next(e)
-                        })
-                    : next()
-                },
               ],
               // 5) done
               (err, results) => {
-                if (err) {
-                  this.reporter.error(this.test, err)
-                  reject(err)
-                } else {
-                  this.reporter.success(this.test)
-                  resolve()
+                const done = () => {
+                  if (err) {
+                    this.reporter.error(this.test, err)
+                    reject(err)
+                  } else {
+                    this.reporter.success(this.test)
+                    resolve()
+                  }
                 }
+
+                // always run test teardown
+                !!this.test.teardown
+                  ? runTeardown(this, this.test.teardown)
+                      .then(res => {
+                        done(res)
+                      })
+                      .catch(e => {
+                        done()
```

Comment:
What if there is not teardown? Does it skips the teardown and ends the test if there is no teardown in the test?

Reply:
Yep, that is what !!this.test.teardown checks.